src/input/datetime.rs

```diff
@@ -291,6 +346,30 @@ pub fn float_as_time<'a>(input: &'a impl Input<'a>, timestamp: f64) -> ValResult
     int_as_time(input, timestamp.floor() as i64, microseconds.round() as u32)
 }
 
+pub fn bytes_as_timedelta<'a, 'b>(input: &'a impl Input<'a>, bytes: &'b [u8]) -> ValResult<'a, EitherTimedelta<'a>> {
+    match Duration::parse_bytes(bytes) {
+        Ok(dt) => Ok(dt.into()),
+        Err(err) => {
+            err_val_error!(
+                input_value = input.as_error_value(),
+                kind = ErrorKind::TimedeltaParsing,
+                context = context!("parsing_error" => err.get_documentation().unwrap_or_default())
+            )
+        }
+    }
+}
+pub fn float_as_duration(total_seconds: f64) -> Duration {
+    let positive = total_seconds >= 0_f64;
+    let total_seconds = total_seconds.abs();
+    let microsecond = total_seconds.fract() * 1_000_000.0;
+    let days = total_seconds as u64 / 86400;
+    let seconds = total_seconds as u64 - days * 86400;
+    Duration::new(positive, days, seconds as u32, microsecond.round() as u32)
+}
+
+pub fn float_as_timedelta<'a>(total_seconds: f64) -> ValResult<'a, EitherTimedelta<'a>> {
```

Comment:
do we really need this method? Could just use `float_as_duration(total_seconds).into()`.